jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -159,6 +170,419 @@ default void clear()
         BufferUtil.clear(getByteBuffer());
     }
 
+    default int append(byte[] bytes, int offset, int length)
+    {
+        ByteBuffer to = getByteBuffer();
+
+        int pos = BufferUtil.flipToFill(to);
+        try
+        {
+            length = Math.min(length, to.remaining());
+            to.put(bytes, offset, length);
+        }
+        finally
+        {
+            BufferUtil.flipToFlush(to, pos);
+        }
+        return length;
+    }
+
+    default void append(ByteBuffer bytes)
+    {
+        BufferUtil.append(getByteBuffer(), bytes);
+    }
+
+    default boolean append(ByteBuffer bytes, Runnable releaser)
```

Comment:
This variant taking a `Runnable releaser` seems at odds with the release doctrine. I'm not sure I like it.

Reply:
I have removed it.  There is a wrapper in RetainableByteBuffer to make one from a releaser.  I agree it is strange, so that keeps the strangeness in one spot.